### PR TITLE
check for overwriting a Hybrid and duplicate ids

### PR DESCRIPTION
**Pull request by mortenjc**
### Issue reference / description

The branch you merge from should already reference an event-formation-unit github ticket number. You can add a descriptive title, but if an issue is referenced, you don't have to.

## Checklist for submitter

- [ ] Check for conflict with integration test
- [ ] Unit tests pass

---

### Nominate for Group Code Review

- [ ] Nominate for code review

### CODE REVIEW

File: src/common/readout/vmm3/VMM3Config.cpp [hunk 1]
```diff
@@ -63,51 +63,55 @@ void VMM3Config::applyVMM3Config() {
...
   } catch (...) {
-    XTRACE(INIT, ERR, "JSON config - error: Invalid Config file: %s",
-           FileName.c_str());
-    LOG(INIT, Sev::Error, "JSON config - error: Invalid Config file: {}",
-        FileName);
-    throw std::runtime_error("Invalid Json file");
+    auto Message = fmt::format("JSON config - error: Invalid Config file: {}",FileName);
+    XTRACE(INIT, ERR, "%s", Message.c_str());
+    LOG(INIT, Sev::Error, Message);
+    throw std::runtime_error("Message");
```
Comment (tbukovics) [inline src/common/readout/vmm3/VMM3Config.cpp:114]: Did you maybe planned to pass the Message variable to the runtime_error instead the "Message" string? I see no reason the throw an error with "Message" error :smiley:

File: src/common/readout/vmm3/VMM3Config.cpp [hunk 2]
```diff
@@ -116,8 +120,9 @@ void VMM3Config::loadAndApplyCalibration(std::string CalibFile) {
   try {
     calib_root = from_json_file(CalibFile);
   } catch (...) {
-    LOG(INIT, Sev::Error, "Error loading json calibration file");
-    throw std::runtime_error("Error loading json calibration file");
+    std::string Message = "Error loading json calibration file";
```
Comment (tbukovics) [inline src/common/readout/vmm3/VMM3Config.cpp:123]: Following the practice above, I would add a parameter to this error message with the name of the file at least.

**Review by tbukovics — CHANGES_REQUESTED** [review]
Hi, first run done, I found some points, which seems incorrect.

File: src/common/readout/vmm3/VMM3Config.cpp (hunk 1 above)
Comment (mortenjc) [reply src/common/readout/vmm3/VMM3Config.cpp:114]: Yes, I thought I had fixed that already. Thanks!

File: src/common/readout/vmm3/VMM3Config.cpp (hunk 2 above)
Comment (mortenjc) [reply src/common/readout/vmm3/VMM3Config.cpp:123]: Agree

**Pull request merged**